torch_geometric/utils/unbatch_edge_index.py

```diff
@@ -0,0 +1,30 @@
+from typing import List
+
+import torch
+from torch import Tensor
+
+from torch_geometric.utils import degree
+
+
+def unbatch_edge_index(edge_index: Tensor, batch: Tensor) -> List[Tensor]:
+    r"""Splits the :obj:`edge_index` according to a :obj:`batch` vector.
+
+    Args:
+        edge_index (Tensor): The edge_index tensor.
+        batch (LongTensor): The batch vector
+            :math:`\mathbf{b} \in {\{ 0, \ldots, B-1\}}^N`, which assigns each
+            node to a specific example. Must be ordered, consecutive, and
+            starts with 0.
+
+    :rtype: :class:`List[Tensor]`
+    """
```

Comment:
I think we can heavily speed this procedure up via:
```python
edge_batch = batch[edge_index[0]]
sizes = degree(edge_batch).cpu().tolist()
return edge_index.split(sizes)
```

Reply:
Yes, you are right. That is clever and removes the need for `torch.bucketize`. But, it imposes new restrictions that the `edge_index` is sorted. Do you think it is okay? In addition, `torch.cumsum` is still necessary since we need to determine the number that has to be subtracted for each graph. I will find a time to benchmark the two versions and let you know.

Reply:
Got it. I think this assumption is reasonable. Let me know how the two versions compare.